test/runify.js

```diff
@@ -56,6 +56,72 @@ hosts.forEach(function (record) {
       }
     });
 
+    describe('child process creation', function() {
+      let agent;
+
+      before(function() {
+        return runify.createAgent(type, options)
+          .then(a => agent = a);
+      });
+
+      after(function() {
+        return agent.destroy();
+      });
+
+      it('forwards hostArgs (via agent.args) correctly', function () {
+        let args = [];
+
+        // The arguments in agent.args are those provided
+        // via the --hostArgs options when called from
+        // test262-harness
+
+        // Setup special cases
+        if (type === 'd8') {
+          agent.args.push('--expose_gc');
+          args.push('-e', 'print(typeof gc === "function");');
+        }
+
+        if (type === 'node') {
+          agent.args.push('--expose_gc');
+          args.push('-e', 'console.log(typeof gc === "function");');
+        }
+
+        if (type === 'jsc') {
+          agent.args.push('--useWebAssembly=true');
+          args.push('-e', 'print(typeof WebAssembly === "function");');
+        }
+
+        if (type === 'jsshell') {
+          agent.args.push('--no-wasm');
+          args.push('-e', 'print(typeof WebAssembly === "undefined");');
+        }
+
+        // chakra cannot be tested in this manner (no -e support),
+        // but the changes apply to that host as well.
+        //
+        // browsers are irrelevant to this test
+        //
+        if (['ch', 'firefox', 'chrome', 'remote'].includes(type)) {
+          this.skip();
+          return;
+        }
+
+        return new Promise(resolve => {
+          let acp = agent.createChildProcess(args);
+          let stdout = '';
+          let stderr = '';
+
+          acp.stdout.on('data', str => { stdout += str; });
+          acp.stderr.on('data', str => { stderr += str; });
+          acp.on('close', () => {
+            acp = null;
+            assert.equal(stdout.trim(), 'true');
```

Comment:
If you assert that `stderr` is the empty string first, then the failure case will be a little more useful (i.e. the unexpected error string will be reported, instead of something like "Expected 'true' but got ''.")